pkg/providers/instancetypes/instancetypes.go

```diff
@@ -68,23 +68,23 @@ type InstanceTypeProvider struct {
 	instanceTypesSeqNum uint64
 }
 
-func NewInstanceTypeProvider(sess *session.Session, ec2api ec2iface.EC2API, subnetProvider *subnet.Provider,
-	unavailableOfferingsCache *awscache.UnavailableOfferings, pricingProvider *pricing.Provider) *InstanceTypeProvider {
-	return &InstanceTypeProvider{
+func NewProvider(sess *session.Session, cache *cache.Cache, ec2api ec2iface.EC2API, subnetProvider *subnet.Provider,
```

Comment:
Is there a reason that we need to take a dependency on the session? Can we just take dependency on the region since this is all that we used?